src/webgpu/api/validation/vertex_state.spec.ts

```diff
@@ -303,73 +310,169 @@ g.test('vertex_attribute_shaderLocation_limit')
     t.testVertexState(success, { vertexBuffers });
   });
 
-g.test('pipeline_vertex_buffers_are_backed_by_attributes_in_vertex_input').fn(async t => {
-  const vertexState: GPUVertexStateDescriptor = {
-    vertexBuffers: [
+g.test('vertex_attribute_shaderLocation_unique')
+  .desc(
+    `Test that shaderLocation must be unique in the vertex state.
+   - Test for various pairs of buffers that contain the potentially conflicting attributes
+   - Test for the potentially conflicting attributes in various places in the buffers (with dummy attributes)
+   - Test for various shaderLocations that conflict or not`
+  )
+  .subcases(() =>
+    params()
+      .combine(poptions('vertexBufferIndexA', [0, 1, kMaxVertexBuffers - 1]))
+      .combine(poptions('vertexBufferIndexB', [0, 1, kMaxVertexBuffers - 1]))
+      .combine(pbool('testAttributeAtStartA'))
+      .combine(pbool('testAttributeAtStartB'))
+      .combine(poptions('shaderLocationA', [0, 1, 7, kMaxVertexAttributes - 1]))
+      .combine(poptions('shaderLocationB', [0, 1, 7, kMaxVertexAttributes - 1]))
+      .combine(poptions('extraAttributes', [0, 4]))
+  )
+  .fn(t => {
+    const {
+      vertexBufferIndexA,
+      vertexBufferIndexB,
+      testAttributeAtStartA,
+      testAttributeAtStartB,
+      shaderLocationA,
+      shaderLocationB,
+      extraAttributes,
+    } = t.params;
+
+    const vertexBufferAttributes: GPUVertexAttributeDescriptor[][] = [];
```

Comment:
Not actually used, could just be `attributesA = [];` `attributesB = [];`

Reply:
Actually we use it so we can have both attributes be aliased or not. Added a comment to clarify this.